File: app/src/main/java/mk/webfactory/template/log/DebugLogger.java
package mk.webfactory.template.log;

import android.util.Log;
import java.util.ArrayList;
import java.util.List;
import timber.log.Timber;

public class DebugLogger extends Timber.DebugTree implements CrashReportLogger {

  private static final String TAG = "CrashReportLogger";

  @Override public void setCurrentPage(String page) {
    log(Log.INFO, TAG, page, null);
  }

  @Override public List<String> getLastUsedPages() {
    return new ArrayList<>();
  }
}

File: app/src/main/java/mk/webfactory/template/log/DebugLogger.kt
package mk.webfactory.template.log

import android.util.Log
import timber.log.Timber.DebugTree
import java.util.*

class DebugLogger : DebugTree(), CrashReportLogger {

    private val lastUsedPages: List<String>
        get() = ArrayList()

    override fun setLoggedInUser(userIdentifier: String) {
        logSetValue("user_id", userIdentifier)
    }

    override fun setCurrentPage(page: String) {
        log(Log.INFO, TAG, page, null)
    }

    override fun lastUsedPages(): List<String> {
        return lastUsedPages
    }

    private fun logSetValue(key: String, value: Any) {
        log(Log.INFO, TAG, "$key=$value", null)
    }

    companion object {
        private const val TAG = "CrashReportLogger"
    }
}
